src/Hl7.Fhir.Core.Tests/Rest/FhirClientTests.cs

```diff
@@ -198,19 +198,56 @@ public void ReadRelativeAsync()
 			Assert.AreEqual("Den Burg", loc.Resource.Address.City);
 		}
 #endif
+        public static void Compression_OnBeforeRequestGZip(object sender, BeforeRequestEventArgs e)
```

Comment:
This is the manual way of setting the headers without using the property.
This way I can test each of the encryption techniques independently.
(otherwise always goes gzip)
Also wanted to ensure that it would decode the content returned, even if it wasn't asked for by the client code.